### PR TITLE
AGS: Update detection_tables.h

### PR DESCRIPTION
**Pull request by berianwilliams**
Bought every commercial AGS game on Steam, scanned and added.

### CODE REVIEW

**Comment by dreammaster** [comment]
AGA will be doing some further additions and fix the commit prefix tomorrow

**Comment by SupSuper** [comment]
There's a bunch of duplicate gameid entries which should be cleaned up.

File: engines/ags/detection_tables.h [hunk 1]
```diff
@@ -1491,36 +1534,83 @@ const AGSGameDescription GAME_DESCRIPTIONS[] = {
...
 	ENGLISH_ENTRY("primordia", "primordia.exe", "f2edc9c3161f1f538df9b4c59fc89e24", 978377182), // Steam
+	ENGLISH_ENTRY("primordia", "primordia.exe", "f2edc9c3161f1f538df9b4c59fc89e24", 978377890), // GOG
 	ENGLISH_ENTRY("puzzlebots", "ac2game.dat", "34b49df9cf6eadb5c3587b3921d5b72f", 787776664), // Steam
 	ENGLISH_ENTRY("resonance", "resonance.exe", "2e635c22bcbf0ed3d46f1bcde71812d4", 849404957), // GOG
 	ENGLISH_ENTRY("resonance", "resonance.exe", "2e635c22bcbf0ed3d46f1bcde71812d4", 849410915), // Steam
+	ENGLISH_ENTRY("richardalice", "richardandalice.exe", "f47d11397e2059a25b11057ea6cf3409", 108264256), // Steam
 	ENGLISH_ENTRY("richardandalice", "richardandalice.exe", "1023d348f67ba958f78ed62d029565c7", 109164768), // Steam
+	ENGLISH_ENTRY("roguestate", "roguestate.exe", "981a1c4d5a64b8ebe300f9769acf0fe5", 895486573), // Steam
 	ENGLISH_ENTRY("shardlight", "Shardlight.exe", "ee801fba52d252249677a9170bd2db96", 642157876), // Steam
+	ENGLISH_ENTRY("shardlight", "shardlight.exe", "ee801fba52d252249677a9170bd2db96", 642157876), // Steam
```
Comment (SupSuper) [inline engines/ags/detection_tables.h:1593]: Duplicate? I think ScummVM is case-insensitive.

**Comment by sev-** [comment]
The commit log message must have an engine prefix. Please see https://wiki.scummvm.org/index.php?title=Commit_Guidelines

File: engines/ags/detection_tables.h (hunk 1 above)
Comment (sev-) [reply engines/ags/detection_tables.h:1593]: Yes, ScummVM is case-insensitive for the file names. Perhaps we would need to improve the scan tool

File: engines/ags/detection_tables.h [hunk 2]
```diff
@@ -1491,36 +1534,83 @@ const AGSGameDescription GAME_DESCRIPTIONS[] = {
...
+	ENGLISH_ENTRY("shivah", "ac2game.dat", "0aaf5445a3544a631d6e7dd4561fc7ae", 32323040), // Steam
+	ENGLISH_ENTRY("shivah", "ac2game.dat", "bdc618b23fc279938f05f8ac058a938b", 32727114), // Steam
```
Comment (sev-) [inline engines/ags/detection_tables.h:1595]: Is there a way to understand the version number, so you could put it in the comment?

Comment (berianwilliams) [reply engines/ags/detection_tables.h:1595]: The AGS editor does allow an author to specify a version number, but I'm not sure whether it's stored in any of the game files' headers?

**Comment by criezy** [comment]
Thank you for all those detection entries.
This pull request was rotting and had numerous conflicts so I manually added your detection manually (and moved some existing detection entries to the commercial game section) in 3310fa27f8.

**Pull request closed without merging**